ConsumerApi.Tests.Integration/StepDefinitions/ChallengesApiStepDefinitions.cs

```diff
@@ -1,49 +1,70 @@
-using Backbone.ConsumerApi.Tests.Integration.API;
+using Backbone.BuildingBlocks.SDK.Endpoints.Common.Types;
+using Backbone.ConsumerApi.Sdk;
+using Backbone.ConsumerApi.Sdk.Authentication;
+using Backbone.ConsumerApi.Sdk.Endpoints.Challenges.Types;
 using Backbone.ConsumerApi.Tests.Integration.Configuration;
-using Backbone.ConsumerApi.Tests.Integration.Models;
-using Backbone.Crypto.Abstractions;
+using Backbone.ConsumerApi.Tests.Integration.Extensions;
+using Backbone.Tooling;
 using Microsoft.Extensions.Options;
-using TechTalk.SpecFlow.Assist;
 
 namespace Backbone.ConsumerApi.Tests.Integration.StepDefinitions;
 
 [Binding]
 [Scope(Feature = "POST Challenge")]
 [Scope(Feature = "GET Challenge")]
-internal class ChallengesApiStepDefinitions : BaseStepDefinitions
+internal class ChallengesApiStepDefinitions
 {
     private string _challengeId;
-    private HttpResponse<Challenge>? _response;
+    private ApiResponse<Challenge>? _response;
+    private Client _sdk;
+    private bool _isAuthenticated;
+    private readonly HttpClient _httpClient;
+    private readonly ClientCredentials _clientCredentials;
 
-    public ChallengesApiStepDefinitions(IOptions<HttpConfiguration> httpConfiguration, ISignatureHelper signatureHelper, ChallengesApi challengesApi, IdentitiesApi identitiesApi, DevicesApi devicesApi) :
-        base(httpConfiguration, signatureHelper, challengesApi, identitiesApi, devicesApi)
+    public ChallengesApiStepDefinitions(IOptions<HttpConfiguration> httpConfiguration, HttpClientFactory factory)
     {
         _challengeId = string.Empty;
+        _httpClient = factory.CreateClient();
+        _clientCredentials = new ClientCredentials(httpConfiguration.Value.ClientCredentials.ClientId, httpConfiguration.Value.ClientCredentials.ClientSecret);
+        _sdk = Client.CreateUnauthenticated(_httpClient, _clientCredentials);
```

Comment:
Is this value ever used? Not really, right? It's overriden in either `AuthenticatedUser` or `GivenTheUserIsUnauthenticated` (depending on which of those the test calls)

Reply:
The only line that could be removed here is the `_sdk` attribution. Even that one is necessary if we wish to allow tests which do not start with `the user is (un)authenticated`. Should this line be removed?

Also, note that it can't really be removed. We cant set `_sdk = null!;` or we can make `_sdk` nullable.

Reply:
> if we wish to allow tests which do not start with the user is (un)authenticated.

This will never be the case, I think.

> Should this line be removed?

Yes.

> we can make _sdk nullable.

Yes, do it.